dealiiadapter.cc

```diff
@@ -993,6 +1011,40 @@ void ElasticProblem<dim>::apply_precice_forces(std::vector<double>& precice_forc
 
 }
 
+template <int dim>
+void ElasticProblem<dim>::save_old_state()
+{
+    // Store current state for implict coupling
+    if (precice.isActionRequired(precice::constants::actionWriteIterationCheckpoint()))
+    {
+        old_state_old_velocity = old_velocity;
+        old_state_old_displacement = old_displacement;
+        old_state_old_forces = old_forces;
+
+        precice.fulfilledAction(precice::constants::actionWriteIterationCheckpoint());
+    }
+
+}
+
+template <int dim>
+void ElasticProblem<dim>::reload_old_state()
+{
+    // Load old state for implicit coupling
+    if (precice.isActionRequired(precice::constants::actionReadIterationCheckpoint()))
+    {
+        velocity = old_velocity;
+        old_velocity = old_state_old_velocity;
+        displacement = old_displacement;
+        old_displacement = old_state_old_displacement;
+        old_forces = old_state_old_forces;
+
+        time.restore();
+
+        precice.fulfilledAction(precice::constants::actionReadIterationCheckpoint());
+    }
+
+}
```

Comment:
Please be aware that this will most probably not work when dealii is subcycling. This is totally OK for the moment, but should be added in the long run. Then, you probably need to store all 5 fields and also the time and timestep. Maybe opening an issue already now would be good such that we don't forget about this.